### ISSUE TITLE
More pointer issues

### ISSUE THREAD

**Issue by kusalananda** [NBISweden/MrBayes#29]
For two statements, one in `utils.c` and one in `model.c`, I get the warning

```
warning: ordered comparison between pointer and zero
```

In `utils.c`:

```
utils.c:5378:49: warning: ordered comparison between pointer and zero
      ('int *' and 'int') is an extension [-Wpedantic]
                if (pt->nEvents[i*pt->nNodes+j] > 0)
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~
```

Looking at what `pt->nEvents[i*pt->nNodes+j]` actually _is_ we see that `pt` is a `PolyTree *`.  `PolyTree` is in turn defined in `bayes.h` as a struct and its `nEvents` member is an `int **`. The comparison is therefore between an `int *` and zero.  _This is a bug_, but I'm not sure what was intended in the code, so I can't fix it.

In `model.c`:

```
model.c:17896:48: warning: ordered comparison between pointer and zero
      ('int *' and 'int') is an extension [-Wpedantic]
        else if (activeParams[P_MIXTURE_RATES] > 0)
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~
```

Here, `activeParams` is a global variable in `model.c` of type `int *[]`.  This means that here, too, we have a comparison between an `int *` and zero. _This is a bug_.

**Comment by kusalananda** [comment]
I'm wondering whether the line in `utils.c`
```
if (pt->nEvents[i*pt->nNodes+j] > 0)
```
should read
```
if (pt->nEvents[i*pt->nNodes][j] > 0)
```

**Comment by andkaha** [comment]
The first warning was fixed with commit 7a4747b29be4a5d8250fc13bbb3ea82a52cbb1a3 (Jan 17, 2018). The second warning was, I believe, fixed with commit dcb23821115caeec91a386955e5e9f10b371eaf2 (Nov 22, 2017). Both by Chi.

Closing this issue.
